src/annotator/guest.js

```diff
@@ -557,7 +557,16 @@ export default class Guest extends Delegator {
       isBackwards
     );
     this.adderCtrl.annotationsForSelection = annotationsForSelection();
-    this.adderCtrl.showAt(left, top, arrowDirection);
+    // Find the zIndex of two close by element and use the greatest.
+    const zIndex1 =
+      findGreatestZindex(document.elementFromPoint(left, top)) + 1;
+    const zIndex2 = findGreatestZindex(selection.focusNode?.parentElement) + 1;
+    this.adderCtrl.showAt(
+      left,
+      top,
+      arrowDirection,
+      Math.max(zIndex1, zIndex2)
+    );
```

Comment:
Can you add a test in `guest-test.js` to cover this change. Here is a sketch of what that could look like:

```diff
diff --git a/src/annotator/test/guest-test.js b/src/annotator/test/guest-test.js
index 8ac1aea26..70e4fb573 100644
--- a/src/annotator/test/guest-test.js
+++ b/src/annotator/test/guest-test.js
@@ -37,6 +37,7 @@ describe('Guest', () => {
   const sandbox = sinon.createSandbox();
   let CrossFrame;
   let fakeCrossFrame;
+  let fakeFindGreatestZindex;
   let highlighter;
   let guestConfig;
   let htmlAnchoring;
@@ -81,10 +82,12 @@ describe('Guest', () => {
     CrossFrame = sandbox.stub().returns(fakeCrossFrame);
     guestConfig.pluginClasses.CrossFrame = CrossFrame;
 
+    fakeFindGreatestZindex = sinon.stub().returns(0);
+
     $imports.$mock({
       './adder': {
         Adder: FakeAdder,
-        findGreatestZindex: sinon.stub().returns(0),
+        findGreatestZindex: fakeFindGreatestZindex,
       },
       './anchoring/html': htmlAnchoring,
       './highlighter': highlighter,
@@ -481,6 +484,22 @@ describe('Guest', () => {
       assert.called(FakeAdder.instance.showAt);
     });
 
+    it('shows the adder above the content', () => {
+      createGuest();
+      fakeFindGreatestZindex.onFirstCall().returns(50);
+      fakeFindGreatestZindex.onSecondCall().returns(10);
+
+      simulateSelectionWithText();
+
+      assert.calledWith(
+        FakeAdder.instance.showAt,
+        sinon.match.number,
+        sinon.match.number,
+        sinon.match.any,
+        50 + 1
+      );
+    });
+
     it('sets the annotations associated with the selection', () => {
       createGuest();
       const ann = {};
```